### PR TITLE
SRU: convert JSON to Dublin Core

### PR DESCRIPTION
**Pull request by rerowep**
* Implements json to doublin core transformation.
* Adds `format=dc` parameter and `application/xml+dc` type for documents search.

Co-Authored-by: Peter Weber <ipeter.weber@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/documents/dojson/contrib/jsontodc/model.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
```
Comment (iGormilhit) [inline rero_ils/modules/documents/dojson/contrib/jsontodc/model.py:4]:
```suggestion
# Copyright (C) 2021 RERO
```

File: rero_ils/modules/documents/dojson/contrib/jsontodc/model.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""rero-ils DoublinCore model definition."""
```
Comment (iGormilhit) [inline rero_ils/modules/documents/dojson/contrib/jsontodc/model.py:18]:
```suggestion
"""RERO ILS Dublin Core model definition."""
```

File: rero_ils/modules/documents/dojson/contrib/jsontodc/model.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+# @dublincore.over('titles', 'title')
+# @utils.ignore_value
+# def json_to_titles(self, key, value):
+#     """Get titles data."""
+#     titles = []
+#     for title in value:
+#         main_titles = title.get('mainTitle', [])
+#         for main_title in main_titles:
+#             titles.append(main_title.get('value'))
+#     return titles', 'None
```
Comment (iGormilhit) [inline rero_ils/modules/documents/dojson/contrib/jsontodc/model.py:108]: Should this be removed?

File: rero_ils/modules/documents/dojson/contrib/jsontodc/model.py [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+# @dublincore.over('rights', '^rights')
+# @utils.ignore_value
+# def json_to_rights(self, key, value):
+#     """Get rights data."""
+
+
+# @dublincore.over('sources', '^sources')
+# @utils.ignore_value
+# def json_to_sources(self, key, value):
+#     """Get sources data."""
```
Comment (iGormilhit) [inline rero_ils/modules/documents/dojson/contrib/jsontodc/model.py:268]: Should this be removed?

File: rero_ils/modules/documents/serializers.py [hunk 5]
```diff
@@ -169,5 +175,153 @@ def _process_library_buckets(cls, org, lib_buckets):
 json_doc = DocumentJSONSerializer(RecordSchemaJSONV1)
 """JSON v1 serializer."""
 
+
+DEFAULT_LANGUAGE = 'en'
+
+
+class DublinCoreSerializer(BaseDublinCoreSerializer):
+    """DublinCore serializer for records.
```
Comment (iGormilhit) [inline rero_ils/modules/documents/serializers.py:183]:
```suggestion
    """Dublin Core serializer for records.
```

File: tests/unit/test_documents_dojson_dc.py [hunk 6]
```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
```
Comment (iGormilhit) [inline tests/unit/test_documents_dojson_dc.py:4]:
```suggestion
# Copyright (C) 2021 RERO
```

File: tests/unit/test_documents_dojson_dc.py [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""DOJSON transiformation for ebooks module tests."""
```
Comment (iGormilhit) [inline tests/unit/test_documents_dojson_dc.py:18]:
```suggestion
"""DOJSON transformation for ebooks module tests."""
```

File: tests/unit/test_documents_dojson_dc.py [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+
+from __future__ import absolute_import, print_function
+
+from flask_babelex import gettext as _
+
+from rero_ils.modules.documents.dojson.contrib.jsontodc import dublincore
+
+
+def test_title_to_dc():
+    """Test title to dublincore transformation."""
```
Comment (iGormilhit) [inline tests/unit/test_documents_dojson_dc.py:28]:
```suggestion
    """Test title transformation to Dublin Core."""
```

File: tests/unit/test_documents_dojson_dc.py [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+    assert result == {'titles': ['Suisse']}
+
+
+# def test_contribution_to_dc():
+#     """Test contribution to dublincore transformation."""
+#     pass
+
+
+def test_summary_note_dissertation_supplementarycontent_to_dc():
+    """Test descriptions transformation."""
```
Comment (iGormilhit) [inline tests/unit/test_documents_dojson_dc.py:96]:
```suggestion
    """Test description transformation to Dublin Core."""
```

File: tests/unit/test_documents_dojson_dc.py [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+        'Fictions jeunesse: roman, jeux vidéo, imagination',
+        'In : Edinburgh medical and surgical journal. - Edinburgh. - '
+        'No. 55(1818), 64 p.',
+        'Bibliogr.: p. 52',
+        'Th. Sc. techn. Lausanne',
+    ]}
+
+
+def test_language_to_dc():
+    """Test language to dublincore transformation."""
```
Comment (iGormilhit) [inline tests/unit/test_documents_dojson_dc.py:123]:
```suggestion
    """Test language transformation to Dublin Core."""
```

File: tests/unit/test_documents_dojson_dc.py [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+            'type': 'bf:Language',
+            'value': 'ain'
+        }
+    ]}
+    result = dublincore.do(record)
+    assert result == {'languages': ['ain']}
+
+
+def test_provision_activity_to_dc():
+    """Test provisionActivity to dublincore transformation."""
```
Comment (iGormilhit) [inline tests/unit/test_documents_dojson_dc.py:135]:
```suggestion
    """Test provisionActivity transformation to Dublin Core ."""
```

File: tests/unit/test_documents_dojson_dc.py [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+            'country': 'xx',
+            'type': 'bf:Place'
+         }]
+    }]}
+    result = dublincore.do(record)
+    assert result == {'dates': ['1978']}
+
+
+def test_type_to_dc():
+    """Test type to dublincore transformation."""
```
Comment (iGormilhit) [inline tests/unit/test_documents_dojson_dc.py:177]:
```suggestion
    """Test type transformation to Dublin Core ."""
```

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
SRUI: convert JSON to Dublin Core

* Implements JSON to Dublin Core transformation.
* Adds `format=dc` parameter and `application/xml+dc` type for documents search.
```

File: rero_ils/modules/documents/dojson/contrib/jsontodc/model.py (hunk 4 above)
Comment (rerowep) [reply rero_ils/modules/documents/dojson/contrib/jsontodc/model.py:268]: Is in specification and should be implemented later.

**Review by zannkukai — APPROVED** [review]

**Review by jma — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]

**Pull request merged**